tests/documentation/command-reference/doc_command_reference_init_test.go

```diff
@@ -0,0 +1,220 @@
+				got = strings.Replace(got, "? Enter component name: (node-echo) ", "", 1)
+				want := helper.GetMDXContent(filepath.Join(commonPath, "interactive_mode_directory_with_sources_output.mdx"))
+				diff := cmp.Diff(want, got)
+				Expect(diff).To(BeEmpty())
+			})
+		})
+	})
+	Context("Non Interactive Mode", func() {
+
+		It("Fetch Devfile of a specific version", func() {
+			args := []string{"init", "--devfile", "go", "--name", "my-go-app", "--devfile-version", "2.0.0"}
+			out := helper.Cmd("odo", args...).ShouldPass().Out()
+			got := fmt.Sprintf(outputStringFormat, strings.Join(args, " "), helper.StripSpinner(out))
+			got = helper.ReplaceTimeInString(got, "[3s]")
+			want := helper.GetMDXContent(filepath.Join(commonPath, "versioned_devfile_output.mdx"))
+			diff := cmp.Diff(want, got)
+			Expect(diff).To(BeEmpty())
+		})
+
+		It("Fetch Devfile of the latest version", func() {
+			args := []string{"init", "--devfile", "go", "--name", "my-go-app", "--devfile-version", "latest"}
+			out := helper.Cmd("odo", args...).ShouldPass().Out()
+			got := fmt.Sprintf(outputStringFormat, strings.Join(args, " "), helper.StripSpinner(out))
+			got = helper.ReplaceTimeInString(got, "[4s]")
+			want := helper.GetMDXContent(filepath.Join(commonPath, "latest_versioned_devfile_output.mdx"))
+			diff := cmp.Diff(want, got)
+			Expect(diff).To(BeEmpty())
+		})
+
+		It("Fetch Devfile from a URL", func() {
+			args := []string{"init", "--devfile-path", "https://registry.devfile.io/devfiles/nodejs-angular", "--name", "my-nodejs-app", "--starter", "nodejs-angular-starter"}
+			out := helper.Cmd("odo", args...).ShouldPass().Out()
+			got := fmt.Sprintf(outputStringFormat, strings.Join(args, " "), helper.StripSpinner(out))
+			got = helper.ReplaceTimeInString(got, "[4s]")
+			want := helper.GetMDXContent(filepath.Join(commonPath, "devfile_from_url_output.mdx"))
+			want = helper.ReplaceTimeInString(want, "[4s]")
+			diff := cmp.Diff(want, got)
+			Expect(diff).To(BeEmpty())
+		})
+
+		Context("fetching devfile from a registry", func() {
+			When("setting up the registry", func() {
+				const (
+					defaultReg    = "DefaultDevfileRegistry"
+					defaultRegURL = "https://registry.devfile.io"
+					stagingReg    = "StagingRegistry"
+					stagingRegURL = "https://registry.stage.devfile.io"
+				)
+				BeforeEach(func() {
+					helper.Cmd("odo", "preference", "remove", "registry", defaultReg, "-f").ShouldPass()
+					helper.Cmd("odo", "preference", "add", "registry", defaultReg, defaultRegURL).ShouldPass()
+
+					helper.Cmd("odo", "preference", "add", "registry", stagingReg, stagingRegURL).ShouldPass()
+				})
+
+				AfterEach(func() {
+					helper.Cmd("odo", "preference", "remove", "registry", stagingReg, "-f").ShouldPass()
+					helper.SetDefaultDevfileRegistryAsStaging()
+				})
+
+				removePreferenceKeys := func(docString string) (returnString string) {
```

Comment:
This seems too much specific to me. It would be nice to have a mechanism to have `[...]` in the MDX file match any number of lines in the output, so thsi could be reused for other tests

Reply:
I had a similar thought, but there aren't many mdx/md files at present that use `[...]`, so I decide to be more specific. I would prefer to iterate over this when the need arises.